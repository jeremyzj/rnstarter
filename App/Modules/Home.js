import Immutable from 'immutable'
import {actionTypes} from '../Networks/NetworkConfig'

export const types = {
  REPO: 'Home.Repo',
  NUMBER: 'Home.Number'
}

export const actions = {
  [types.REPO] :() =>  ({
    type: types.REPO,
    path: 'user/repos',
    requestType: actionTypes.GET,
    key: 'repos'
  }),
  [types.NUMBER]:(number) => ({
    type: types.NUMBER,
    payload: number,
    key: 'number'
  })
}


const initialState = Immutable.fromJS({
  repos: [],
  number : 0
})

export default (state = initialState, action) => (actions[action.type]
  ? state.set(action.key, action.payload)
  : state)
