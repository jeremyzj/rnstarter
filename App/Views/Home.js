import React, {Component} from 'react'
import {View, Text} from 'react-native'

export default class Profile extends Component {
  render() {
    return (
      <View><Text>Home</Text></View>
    )
  }
}