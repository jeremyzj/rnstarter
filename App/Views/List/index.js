import React, {Component} from 'react'
import {View, Text, Image} from 'react-native'
import {ListFocusImg , ListNormalImg} from '../../Images'

export default class List extends Component {

  static navigationOptions = {
    title: 'åˆ—è¡¨',
    tabBarIcon: ({ focused }) => <Image source={focused ? ListFocusImg : ListNormalImg} style={[{ marginBottom: 1 }]} />,      
  }

  render() {
    return (
      <View><Text>List</Text></View>
    )
  }
}