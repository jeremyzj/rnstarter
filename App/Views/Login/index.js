import React, {Component} from 'react'
import {View, Text} from 'react-native'

export default class Profile extends Component {

  static navigationOptions = {
    title: '登录'
  }

  render() {
    return (
      <View><Text>Login</Text></View>
    )
  }
}