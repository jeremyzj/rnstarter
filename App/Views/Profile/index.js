import React, {Component} from 'react'
import {View, Text, Button, Image, StyleSheet} from 'react-native'
import ProfileFocusImg from '../../Images/profile_focus.png'
import ProfileNormalImg from '../../Images/profile_normal.png'
import { connect } from 'react-redux'

const mapStateToProps = () => ({
})

const mapDispatchToProps = () => ({
  
})

@connect(mapStateToProps, mapDispatchToProps)
export default class Profile extends Component {

  static navigationOptions = {
    title: '个人中心',
    tabBarIcon: ({ focused }) => <Image source={focused ? ProfileFocusImg : ProfileNormalImg} style={[{ marginBottom: 1 }]} />,          
  }

  _login() {
    const {navigation} = this.props
    navigation && navigation.navigate('Login')
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>Profile</Text>
        <Button title="log in" onPress={()=>{this._login()}}/>
      </View>
    )
  }

}

let styles = StyleSheet.create({
  container: {
    flex: 1
  }
})