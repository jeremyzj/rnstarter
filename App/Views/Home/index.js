import React, {Component} from 'react'
import {View, Text, Image, Button} from 'react-native'
import { connect } from 'react-redux'
import * as HomeModules from '../../Modules/Home'
import {HomeNormalImg, HomeFocusImg} from '../../Images'

const mapStateToProps = (state) => ({
  repos: state.getIn(['home', 'repos']),
  number: state.getIn(['home', 'number'])
})

const mapDispatchToProps = (dispatch) => ({
  requestRepo: () => dispatch(HomeModules.actions[HomeModules.types.REPO]()),
  increase: (number) => dispatch(HomeModules.actions[HomeModules.types.NUMBER](number + 1)),
  decrease: (number) => dispatch(HomeModules.actions[HomeModules.types.NUMBER](number - 1)),
})

export class Home extends Component {
  static navigationOptions = {
    title: '首页',
    tabBarIcon: ({ focused }) => <Image source={focused ? HomeFocusImg : HomeNormalImg} style={[{ marginBottom: 1 }]} />,   
  }

  _request() {
    this.props.requestRepo()
  }

  _increase() {
    const { number } = this.props
    this.props.increase(number)
  }

  _decrease() {
    const { number } = this.props
    this.props.decrease(number)
  }

  render() {
    const { number } = this.props

    return (
      <View>
        <Text>Home</Text>
        <Button title="加" onPress={this._increase.bind(this)}/> 
        <Button title="减" onPress={this._decrease.bind(this)}/> 
        <Text>{number}</Text>
        <Button title="request" onPress={this._request.bind(this)}/>  
      </View>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)