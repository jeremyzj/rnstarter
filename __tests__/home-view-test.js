import 'react-native'
import React from 'react'
import {Home} from '../App/Views/Home'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON()
  expect(tree).toMatchSnapshot()
})