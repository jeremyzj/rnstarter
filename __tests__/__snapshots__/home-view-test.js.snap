// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        加
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        减
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        request
      </Text>
    </View>
  </View>
</View>
`;
